test/server/cards/01_SOR/events/MakeAnOpening.spec.js

```diff
@@ -0,0 +1,37 @@
+describe('Make an Opening', function () {
+    integration(function () {
+        describe('Make an Opening\'s ability', function () {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['make-an-opening'],
+                        groundArena: ['pyke-sentinel', { card: 'gideon-hask#ruthless-loyalist', upgrades: ['entrenched'] }],
+                    },
+                    player2: {
+                        groundArena: ['atst', 'isb-agent'],
+                        spaceArena: ['cartel-spacer'],
+                        leader: { card: 'sabine-wren#galvanized-revolutionary', deployed: true }
+                    }
+                });
+            });
+
+            it('should downgrade an unit and heal base', function () {
+                this.player1.clickCard(this.makeAnOpening);
+                expect(this.player1).toBeAbleToSelectExactly([this.pykeSentinel, this.gideonHask, this.atst, this.isbAgent, this.cartelSpacer, this.sabineWren]);
+
+                this.player1.clickCard(this.isbAgent);
+                // VBL : maybe a bug somewhere, if I keep this test here he fails, but at the end of event it works
```

Comment:
This is intended behavior, none of the card effects resolve until all targets are selected. However, as I mentioned in the comment on the implementation, the player should not be required to choose a target for the other effect since it is known in advance ("your base")